src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,258 @@
+import json
+
+from typing import Any, Dict, List, Optional
+
+from loguru import logger
+from sqlalchemy import and_
+from sqlalchemy.orm import Query, Session
+
+from fides.api.common_exceptions import MessageDispatchException
+from fides.api.graph.config import GraphDataset, CollectionAddress
+from fides.api.graph.execution import ExecutionNode
+from fides.api.models.connectionconfig import ConnectionConfig
+from fides.api.models.datasetconfig import DatasetConfig
+from fides.api.models.privacy_request import (
+    ExecutionLog,
+    ExecutionLogStatus,
+    RequestTask,
+    PrivacyRequest,
+    TraversalDetails,
+)
+from fides.api.models.policy import Policy
+from fides.api.schemas.connection_configuration.connection_secrets_dynamic_erasure_email import (
+    DynamicErasureEmailSchema,
+)
+from fides.api.schemas.policy import ActionType
+from fides.api.service.connectors.base_connector import BaseConnector
+from fides.api.service.connectors.erasure_email_connector import (
+    GenericErasureEmailConnector,
+    filter_user_identities_for_connector,
+    send_single_erasure_email,
+)
+from fides.config import get_config
+
+CONFIG = get_config()
+
+
+class DynamicErasureEmailConnector(GenericErasureEmailConnector):
+    """Email Erasure Connector that performs a lookup for the email to use."""
+
+    config_schema = DynamicErasureEmailSchema
+
+    def __init__(self, configuration: ConnectionConfig):
+        super().__init__(configuration)
+
+    def get_email_address_from_custom_request_fields(
```

Comment:
hopefully the docstrings and comments make this function self-explanatory, but this is where the main logic of this feature lives. We use the graph_dataset generated from the dataset reference to get the relevant connector and execute a standalone query against it in order to retireve the email address using the custom fields of the privacy request

Reply:
If we have the nodes hooked up properly, then we wouldn't need a "standalone" query - the custom fields connector could have already retrieved the relevant row and stored it on the request task before we got to the dynamic erasure  email connector